process/block/preprocess/validatorInfoPreProcessor.go

```diff
@@ -0,0 +1,148 @@
+package preprocess
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+type validatorInfoPreprocessor struct {
+	hasher      hashing.Hasher
+	marshalizer marshal.Marshalizer
+}
+
+// NewValidatorInfoPreprocessor creates a new validatorInfo preprocessor object
+func NewValidatorInfoPreprocessor(
+	hasher hashing.Hasher,
+	marshalizer marshal.Marshalizer,
+) (*validatorInfoPreprocessor, error) {
+	if check.IfNil(hasher) {
+		return nil, process.ErrNilHasher
+	}
+	if check.IfNil(marshalizer) {
+		return nil, process.ErrNilMarshalizer
+	}
+
+	rtp := &validatorInfoPreprocessor{
+		hasher:      hasher,
+		marshalizer: marshalizer,
+	}
+	return rtp, nil
+}
+
+// IsDataPrepared returns nil because the TxHashes are the actual serialized objects neeed
+func (rtp *validatorInfoPreprocessor) IsDataPrepared(_ int, _ func() time.Duration) error {
+	return nil
+}
+
+// RemoveTxBlockFromPools removes reward transactions and miniblocks from associated pools
+func (rtp *validatorInfoPreprocessor) RemoveTxBlockFromPools(body block.Body, miniBlockPool storage.Cacher) error {
+	return nil
+}
+
+// RestoreTxBlockIntoPools restores the reward transactions and miniblocks to associated pools
+func (rtp *validatorInfoPreprocessor) RestoreTxBlockIntoPools(
+	body block.Body,
+	miniBlockPool storage.Cacher,
+) (int, error) {
+	if miniBlockPool == nil {
+		return 0, process.ErrNilMiniBlockPool
+	}
+
+	rewardTxsRestored := 0
+	for i := 0; i < len(body); i++ {
+		miniBlock := body[i]
+		if miniBlock.Type != block.PeerBlock {
+			continue
+		}
+
+		miniBlockHash, err := core.CalculateHash(rtp.marshalizer, rtp.hasher, miniBlock)
+		if err != nil {
+			return rewardTxsRestored, err
+		}
+
+		miniBlockPool.Put(miniBlockHash, miniBlock)
+
+		rewardTxsRestored += len(miniBlock.TxHashes)
+	}
+
+	return rewardTxsRestored, nil
+
+	return 0, nil
+}
+
+// ProcessBlockTransactions processes all the reward transactions from the block.Body, updates the state
+func (rtp *validatorInfoPreprocessor) ProcessBlockTransactions(
+	body block.Body,
+	haveTime func() bool,
+) error {
+	return nil
```

Comment:
Missing functionality?

Reply:
will come in the shard process  PR. Added TODO on validatorInfoPreProcessor